homeassistant/components/upnp/device.py

```diff
@@ -32,6 +33,20 @@
 )
 
 
+async def async_get_mac_address_from_host(hass: HomeAssistant, host: str) -> str | None:
+    """Get mac address from host."""
+    ip_addr = ip_address(host)
+    if ip_addr.version == 4:
+        mac_address = await hass.async_add_executor_job(
+            partial(get_mac_address, ip=host)
+        )
+    else:
+        mac_address = await hass.async_add_executor_job(
+            partial(get_mac_address, ip6=host)
+        )
```

Comment:
I seem to recall that `get_mac_address` is expected to throw a few different exceptions. Do we need to trap any of them here?

Reply:
Disregard. The library now states `Exceptions will be handled silently and returned as a None.`. I'm pretty sure it was a bug in an old version